File: examples/Haskell/hwclient.hs
{-# LANGUAGE OverloadedStrings #-}

-- Hello World client

module Main where

import Control.Monad
import System.ZMQ4.Monadic

main :: IO ()
main = runZMQ $ do
    liftIO $ putStrLn "Connecting to hello world server..."

    requester <- socket Req
    connect requester "tcp://localhost:5555"

    forM_ [1..10] $ \i -> do
        liftIO . putStrLn $ "Sending Hello " ++ show i ++ "..."
        send requester [] "Hello"
        _ <- receive requester
        liftIO . putStrLn $ "Received World " ++ show i

File: examples/OCaml/hwclient.ml
(** Hello World client *)

open Zmq
open Helpers

let () =
  printfn "Connecting to hello world server...";
  with_context @@ fun ctx ->
  with_socket ctx Socket.req @@ fun req ->
  Socket.connect req "tcp://localhost:5555";
  for i = 0 to 9 do
    printfn "Sending Hello %d..." i;
    Socket.send req "Hello";
    let answer = Socket.recv req in
    printfn "Received %d : %S" i answer
  done
